### PR TITLE
[#175476527] release pipeline

### PR DESCRIPTION
**Pull request by balanza**
Same changes applied in https://github.com/pagopa/io-functions-services/pull/96.

### Changes in brief
* Introduce the `Release` stage on the pipeline which marks a release on the repository
* Refactor the whole CI/CD to have 
  * separated pipelines for code review and deploy scenarios
  * everything is into `.devops` folder now

### CODE REVIEW

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                !docker-compose.yml
+                !Dangerfile.ts
+                !jest.config.js
+                !local.settings.json
+                !test
+                !tslint.json
+                !tsconfig.json
+            displayName: 'Copy deploy files'
+            
+          - task: AzureFunctionApp@1
```
Comment (gunzip) [inline .devops/deploy-pipelines.yml:124]: this is not a function, it's an app service, is this correct?

Comment (gunzip) [reply .devops/deploy-pipelines.yml:124]: anyway keep this at very lwo priority :)

Comment (balanza) [reply .devops/deploy-pipelines.yml:124]: yes, let me review as I copied the old files

Comment (balanza) [reply .devops/deploy-pipelines.yml]: ok done

**Review by pasqualedevita — APPROVED** [review]

File: .devops/azure-templates/make-build-steps.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  displayName: 'Checkout'
+
+- task: Cache@2
+  inputs:
+    key: 'yarn-${{ parameters.cache_version_id }} | "$(Agent.OS)" | yarn.lock'
+    restoreKeys: |
+      yarn-${{ parameters.cache_version_id }} | "$(Agent.OS)"
+    path: $(YARN_CACHE_FOLDER)
+  displayName: 'Cache yarn packages'
+
```
Comment (pasqualedevita) [inline .devops/azure-templates/make-build-steps.yml:32]: add this 
```suggestion
cacheHitVar: CACHE_RESTORED
```
here why https://github.com/pagopa/azure-pipeline-templates/pull/2#issuecomment-734829613

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Review by pasqualedevita — APPROVED** [review]
no healthcheck

**Comment by balanza** [comment]
> no healthcheck

There's no such feature in the project, yet

**Pull request merged**